src/main/scala/ai/starlake/schema/model/Metadata.scala

```diff
@@ -192,17 +193,15 @@ case class Metadata(
   @JsonIgnore
   def getWrite(): WriteMode = writeStrategy.map(_.getWriteMode()).getOrElse(APPEND)
 
-  @JsonIgnore
   // scalastyle:off null
   def getNullValue(): String = nullValue.getOrElse(if (isEmptyIsNull()) "" else null)
   // scalastyle:on null
 
   @JsonIgnore
-  def getPartitionAttributes()(implicit settings: Settings): List[String] = {
+  def getPartitionAttributes()(implicit settings: Settings): Seq[String] = {
     this.getSink().toAllSinks().partition.getOrElse(Nil)
   }
 
-  @JsonIgnore
```

Comment:
We need to kee this or else Jackson on write would generate this field

Reply:
As discussed, removal of JsonIgnore is kept because it is a value that can be filled by the end user. If we don't plan to change the client value, I would suggest to rename all fields acting like this (and I think there is some more) as getEffective* with a @JsonIgnore on it.

Therefore, we don't change the user input on serialization but we get the value to use in our process. What do you think ?